script/dashboard/dashboard.js

```diff
@@ -28,38 +28,36 @@ app.controller('customerPromisingCtrl', function($scope, $http, customersPromisi
     }
 
     $scope.getColor = function(ratio){
-        if(ratio>1) return 'c-table__ratio1';
-        else if(ratio>0.5) return 'c-table__ratio2';
-        else if(ratio>0.4) return 'c-table__ratio3';
-        else if(ratio>0.3) return 'c-table__ratio4';
-        else if(ratio>0.2) return 'c-table__ratio5';
-        else if(ratio<0.009) return 'c-table__ratio10';
-        else if(ratio<0.09) return 'c-table__ratio9';
-        else if(ratio<0.05) return 'c-table__ratio8';
-        else if(ratio<0.9) return 'c-table__ratio7';
-        else if(ratio<0) return 'c-table__ratio6';
+       return colorService.getColor(ratio);
     }
 });
 
-app.controller('popularProductsCtrl', function($scope, $http){
+app.controller('popularProductsCtrl', function($scope, $http, colorService){
     $http.get('http://localhost:8080/product/popular')
         .then(function(result){
             $scope.popularProducts = result.data;
         });
 
-    $scope.predicate = 'productId';
+    $scope.predicate = '';

```

Comment:
tak, ponieważ ja już tobie zwracam posortowane produkty  z backendu, a Ty od strony frontu na dzień dobry je sortujesz po id :) Wywyaliłem tylko to sortowanie początkowe, normalnie jak klikniesz to bedzie sortowało po czym tam chcesz :)